packages/backend/src/tools/updateDiffHistory.ts

```diff
@@ -45,14 +45,25 @@ async function updateDiffHistoryFile() {
   )
 
   if (diff.length > 0) {
-    const newHistoryEntry = generateDiffHistoryMarkdown(diff, mainBranchHash)
     const diffHistoryPath = `${discoveryFolder}/diffHistory.md`
+    const diskDiffHistory = readFileSync(diffHistoryPath, 'utf-8')
```

Comment:
Program breaks here if there is no existing `diffHistory.md` file. Please cover this scenario.